vertx-web/src/main/java/io/vertx/ext/web/handler/impl/FormLoginHandlerImpl.java

```diff
@@ -78,56 +82,51 @@ public FormLoginHandlerImpl(AuthenticationProvider authProvider, String username
   }
 
   @Override
-  public void handle(RoutingContext context) {
+  public void parseCredentials(RoutingContext context, Handler<AsyncResult<Credentials>> handler) {
     HttpServerRequest req = context.request();
     if (req.method() != HttpMethod.POST) {
-      context.fail(405); // Must be a POST
+      handler.handle(Future.failedFuture(BAD_METHOD)); // Must be a POST
     } else {
       if (!req.isExpectMultipart()) {
```

Comment:
Here we can improve this, replace the current if with:

```java
if (!((RoutingContextInternal) context).seenHandler(RoutingContextInternal.BODY_HANDLER)) {
  context.fail(500, new NoStackTraceThrowable("BodyHandler is required to process POST requests"));
} else {
 ...
```

So this way we are sure that the body handler was used and we can remove all the logs messages and hints

Reply:
As this block of code lies on the `parseCredentials(RoutingContext context, Handler<AsyncResult<Credentials>> handler)` which requires the exception to be propagated via the handler, so I update it to:
```java
      if (!((RoutingContextInternal) context).seenHandler(RoutingContextInternal.BODY_HANDLER)) {
        handler.handle(Future.failedFuture(new NoStackTraceThrowable("BodyHandler is required to process POST requests")));
      }
```